git-link.el

```diff
@@ -411,12 +413,30 @@ return (FILENAME . REVISION) otherwise nil."
                                 (format "L%s-L%s" start end)
                               (format "L%s" start)))))))
 
+(defun git-link-azure (hostname dirname filename branch commit start end)
+  (format "https://%s/%s?path=%s&version=%s%s"
+	  hostname
+	  dirname
+      filename
+      (concat "G" (if branch "B" "C") (or branch commit))
+      (concat
+       (when start (format "&line=%s" start))
+       (when end (format "&lineEnd=%s" end)))))
+
 (defun git-link-commit-github (hostname dirname commit)
   (format "https://%s/%s/commit/%s"
 	  hostname
 	  dirname
 	  commit))
 
+(defun git-link-commit-azure (hostname dirname commit)
+  (format "https://%s/%s/commit/%s"
+	  hostname
+	  dirname
+	  (with-temp-buffer
+        (vc-git--out-ok "rev-parse" commit)
```

Comment:
What's the reason for calling `vc-git--out-ok` here? This is not done elsewhere in the existing code and it (seems) to work fine without it when linking to commits.

Reply:
Also given emacs conventions `vc-git--out-ok` is not part of the public API.

Reply:
I use it to get the full commit hash as the Azure interface seem not to handle short commits well.

I usually run git-link from the magit log buffer. The hash that appear in that buffer is short.

Do you have another suggestion for getting the full commit hash?

Reply:
> Do you have another suggestion for getting the full commit hash?

I will investigate.

Reply:
Thinking back and looking though Emacs' docs –and this code– I recall that at one point I thought: "I should have just used Emacs's `vc-xxx` [or whatever] functions to interact with git" but then I found that there isn't a well-defined public API for this type of usage.   

The way I have done things like this is to make a function to call `git`. There are [several examples of this](https://github.com/sshaw/git-link/blob/5581b4760aa2efb01eddca035a2758bc62da8635/git-link.el#L209) in the code now.